interpreter/runtime/table.ml

```diff
@@ -21,7 +21,11 @@ let valid_limits {min; max} =
 
 let create size r =
   try Lib.Array32.make size r
-  with Out_of_memory | Invalid_argument _ -> raise OutOfMemory
+  with
+  | Out_of_memory -> raise OutOfMemory
+  | Invalid_argument _ ->
+      (* TODO: why are we raising OutOfMemory here ? *)
```

Comment:
> From Wasm's perspective the argument is considered unsigned, so any value interpreted as negative down below is indistinguishable from "way too large".

OK then it may be clearer to use something like `Int64.unsigned_to_int n` in `Array32.make` from `interpreter/util/lib.ml` before bounds checks and remove this `Invalid_argument` catch ?

But OK anyway, I'll leave it as is for now.

> The existing check in Array32 was more intended to capture possible overflows in the conversion from i32 to nativeint. Does Ocaml not raise Invalid_arg there either? If not, I'd leave that alone in order to not hardcode / duplicate implementation specifics. Otherwise, feel free to change it.

When using `Int32.to_int` I don't believe an `Invalid_argument` is raised when `nativeint` is on 31 bits. The doc says:

> Convert the given 32-bit integer (type int32 ) to an integer (type int ).  On 32-bit platforms,  the 32-bit integer is taken modulo 2^31, i.e. the high-order bit is lost during the conversion.  On 64-bit platforms, the conversion is exact.